cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -83,9 +86,10 @@ public void validateProperties(Map<String, String> properties) {
     if (privateInstance && conf.isPreferExternalIP()) {
       // When prefer external IP is set to true it means only Dataproc external ip can be used to for communication
       // the instance being private instance is incapable of using external ip for communication
-      throw new IllegalArgumentException("The instance is incapable of using external ip for communication " +
+      throw new InvalidConfigurationException("The instance is incapable of using external ip for communication " +
```

Comment:
this specific exception class looks like it should always set the error group to configuration instead of having it passed in.

Reply:
The rule engine that categorizes errors has access to the exception as well. So it already knows there is a configuration error. So I thought I will use the ErrorGroup to give additional context on which category of configuration.

Reply:
This code has been removed.